ledger/txtail.go

```diff
@@ -178,21 +316,21 @@ func (t *txTail) produceCommittingTask(committedRound basics.Round, dbRound basi
 	return dcr
 }
 
-// txtailMissingRound is returned by checkDup when requested for a round number below the low watermark
-type txtailMissingRound struct {
+// errTxTailMissingRound is returned by checkDup when requested for a round number below the low watermark
+type errTxTailMissingRound struct {
 	round basics.Round
 }
 
 // Error satisfies builtin interface `error`
-func (t txtailMissingRound) Error() string {
+func (t errTxTailMissingRound) Error() string {
 	return fmt.Sprintf("txTail: tried to check for dup in missing round %d", t.round)
 }
 
 // checkDup test to see if the given transaction id/lease already exists. It returns nil if neither exists, or
 // TransactionInLedgerError / LeaseInLedgerError respectively.
 func (t *txTail) checkDup(proto config.ConsensusParams, current basics.Round, firstValid basics.Round, lastValid basics.Round, txid transactions.Txid, txl ledgercore.Txlease) error {
 	if lastValid < t.lowWaterMark {
```

Comment:
```suggestion
	if lastValid <= t.lowWaterMark {
```

Reply:
no, lowWaterMark is incremented at
```
	for ; t.lowWaterMark < rnd; t.lowWaterMark++ {
		delete(t.lastValid, t.lowWaterMark)
	}
```
and it's last value is after the increment becomes rnd, and the loop body is not executed, i.e. does have values with `r == t.lowWaterMark`. That's why less < is used in the condition above